### PR TITLE
update pattern for dataflow job id extraction

### PR DESCRIPTION
**Pull request by lukas-mi**
Dataflow job id is extracted from the logged output of `java` process that starts the Dataflow job, for example, in case of `BeamRunJavaPipelineOperator`.

Currently job id pattern matches characters until first `"` or `\n` is encountered, which is fine for a following case:
- logged line: `[2024-08-27 11:20:22,094] INFO  Submitted job: 2024-08-27_04_20_21-7947372725816706151`
- extracted job id: `2024-08-27_04_20_21-7947372725816706151`

However, if the logger is configured differently, for example, has a whitespace and a suffix at the end with additional information, the pattern extracts the id together with the suffix:
- logged line: `[2024-08-27 11:20:22,094] INFO  Submitted job: 2024-08-27_04_20_21-7947372725816706151 (org.apache.beam.runners.dataflow.DataflowRunner) (main)`
- extracted job id: `2024-08-27_04_20_21-7947372725816706151 (org.apache.beam.runners.dataflow.DataflowRunner) (main)` 

In the previous example suffix `(org.apache.beam.runners.dataflow.DataflowRunner) (main)` should not be extracted as part of the job id.

I updated the pattern by adding the whitespace character `\s` (along side existing `"` and `\n`), indicating the end of job id.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/google/cloud/hooks/dataflow.py
Comment (VladaZakharova) [inline airflow/providers/google/cloud/hooks/dataflow.py:74]: Looks great! Can we add a unit test for this?

**Review by VladaZakharova — CHANGES_REQUESTED** [review]

File: airflow/providers/google/cloud/hooks/dataflow.py
Comment (lukas-mi) [reply airflow/providers/google/cloud/hooks/dataflow.py:74]: I added another test example in job id extraction test

Comment (VladaZakharova) [reply airflow/providers/google/cloud/hooks/dataflow.py:74]: thank you!

**Review by VladaZakharova — APPROVED** [review]

**Comment by lukas-mi** [comment]
@VladaZakharova when will this be merged? :)

**Review by potiuk — APPROVED** [review]

**Comment by VladaZakharova** [comment]
Hi @potiuk ! Can you please merge it?

**Comment by potiuk** [comment]
> @VladaZakharova when will this be merged? :)

When the test pass and someone will merge it.

Since you are the first time contributor - we have to manually approve workflows to see if tests pass, then you have to fix them if they don't. but when you submit new version you will have to wait for someone to see it and approve it (you can ask in general without mentioning anyone to approve your workflows) to signal that you **think** you fixed all the tests.

Also see the contribution docs that explain the process https://github.com/apache/airflow/tree/main/contributing-docs

**Pull request merged**
